server/src/main/java/com/hedera/block/server/mediator/LiveStreamMediatorImpl.java

```diff
@@ -87,24 +90,30 @@ class LiveStreamMediatorImpl extends SubscriptionHandlerBase<SubscribeStreamResp
 
     /**
      * Publishes the given block item to all subscribers. If an exception occurs while persisting
-     * the block item, the service status is set to not running, and all downstream consumers are
+     * the block items, the service status is set to not running, and all downstream consumers are
      * unsubscribed.
      *
-     * @param blockItem the block item from the upstream producer to publish to downstream consumers
+     * @param blockItems the block item from the upstream producer to publish to downstream
+     *     consumers
      */
     @Override
-    public void publish(@NonNull final BlockItem blockItem) {
+    public void publish(@NonNull final List<BlockItem> blockItems) {
 
         if (serviceStatus.isRunning()) {
 
             // Publish the block for all subscribers to receive
-            LOGGER.log(DEBUG, "Publishing BlockItem: " + blockItem);
+            LOGGER.log(DEBUG, "Publishing BlockItem");
+            final SubscribeStreamResponseSet blockItemsSet =
+                    SubscribeStreamResponseSet.newBuilder().blockItems(blockItems).build();
             final var subscribeStreamResponse =
-                    SubscribeStreamResponse.newBuilder().blockItem(blockItem).build();
+                    SubscribeStreamResponse.newBuilder().blockItems(blockItemsSet).build();
             ringBuffer.publishEvent((event, sequence) -> event.set(subscribeStreamResponse));
 
             // Increment the block item counter
-            metricsService.get(LiveBlockItems).increment();
+            Counter liveBlockItems = metricsService.get(LiveBlockItems);
+            for (int i = 0; i < blockItems.size(); i++) {
```

Comment:
This seems like a loop inside a hot path. Do you think that it can be avoided somehow ?

Reply:
This is another example of PlatformSDK metrics API feature limitations, any other prometheus client would have a `increment(number)` overload that increments more than `1` at the time such counter.

This might be fixed in the future by changes to the SDK or we can decide to change the Metrics, however for this refactor I deemed the best course of action to keep the same Metric intention (block items published) or on a follow-up ticket we can decide to get rid of that metric, and focus on another metric that makes more sense.

But for now I did not found a way to increment the metric other than 1 int at a time.

Reply:
This compiles: ` metricsService.get(LiveBlockItems).add(blockItems.size());` but I haven't tested it.

Reply:
I worked!! thanks!! :)

I guess I missed that api due to expecting it as a overload 😅

Reply:
✅ Done!